plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/generator/SQLGeneratorJoin.java

```diff
@@ -66,6 +73,12 @@ public void run(DBRProgressMonitor monitor) throws InvocationTargetException {
         result = sql.toString();
     }
 
+    private String generateTableAlias(DBSEntity entity) {
```

Comment:
Why keep `aliases` a a class variable? It is used only in one function.
As well as dialect.

Just pass them as parameters.